nifi-nar-bundles/nifi-rules-action-handler-bundle/nifi-rules-action-handler-service/src/main/java/org/apache/nifi/rules/handlers/AlertHandler.java

```diff
@@ -0,0 +1,130 @@
+import org.apache.nifi.rules.Action;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+@Tags({"rules", "rules engine", "action", "action handler", "logging", "alerts", "bulletins"})
+@CapabilityDescription("Creates alerts as bulletins based on a provided action (usually created by a rules engine) ")
+public class AlertHandler extends AbstractActionHandlerService {
+
+    public static final PropertyDescriptor DEFAULT_LOG_LEVEL = new PropertyDescriptor.Builder()
+            .name("Log Level")
+            .required(true)
+            .description("The Log Level to use when logging the Attributes")
+            .allowableValues(DebugLevels.values())
+            .defaultValue("info")
+            .build();
+
+    public static final PropertyDescriptor DEFAULT_CATEGORY = new PropertyDescriptor.Builder()
+            .name("Alert Category")
+            .required(true)
+            .description("The Log Level to use when logging the Attributes")
+            .defaultValue("Rules Triggered Alert")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor DEFAULT_MESSAGE = new PropertyDescriptor.Builder()
+            .name("Alert Message")
+            .required(true)
+            .description("The default message to include in alert")
+            .defaultValue("An alert was triggered by a rules based action.")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    private List<PropertyDescriptor> properties;
+    private String defaultCategory;
+    private String defaultLogLevel;
+    private String defaultMessage;
+
+    @Override
+    protected void init(ControllerServiceInitializationContext config) throws InitializationException {
+        super.init(config);
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(DEFAULT_LOG_LEVEL);
+        properties.add(DEFAULT_CATEGORY);
+        properties.add(DEFAULT_MESSAGE);
+        this.properties = Collections.unmodifiableList(properties);
+    }
+
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) throws InitializationException {
+        defaultLogLevel = context.getProperty(DEFAULT_LOG_LEVEL).getValue().toUpperCase();
+        defaultCategory = context.getProperty(DEFAULT_CATEGORY).getValue();
+        defaultMessage = context.getProperty(DEFAULT_MESSAGE).getValue();
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public void execute(Action action, Map<String, Object> facts) {
+        throw new UnsupportedOperationException("This method is not supported.  The AlertHandler requires a Reporting Context");
+    }
+
+    @Override
+    public void execute(PropertyContext propertyContext, Action action, Map<String, Object> facts) {
```

Comment:
Just want to confirm that `facts` is not applicable in this situation?

Reply:
You know I do need to add an option to allow inclusion of facts in bulletin (similar to the log handler). I'll take care of this.